boards/cats_rev1Pro/Core/Src/control/orientation_kf.c

```diff
@@ -162,20 +163,85 @@ void normalize_q(float32_t* input) {
   input[3] /= abs_value;
 }
 
-void compute_angle(imu_data_t* data, orientation_filter_t* filter) {
-  /* Scale acceleration and rotate into inertial frame */
-  float32_t quat_meas[4] = {(float32_t)(data->acc_z) / 1024.0f, -(float32_t)(data->acc_y) / 1024.0f,
-                            (float32_t)(data->acc_x) / 1024.0f, 0};
-  float32_t abs_sq = quat_meas[0] * quat_meas[0] + quat_meas[1] * quat_meas[1] + quat_meas[2] * quat_meas[2];
-  float32_t abs = 0;
-  arm_sqrt_f32(abs_sq, &abs);
-
-  quat_meas[0] /= abs;
-  quat_meas[1] /= abs;
-  quat_meas[2] /= abs;
-
+void compute_angle(imu_data_t* data, magneto_data_t* magneto_data, orientation_filter_t* filter) {
+  /* Preprocess IMU Data */
+  float32_t scaled_imu[3] = {((float32_t)(data->acc_x)) / 1024.0f, ((float32_t)(data->acc_y)) / 1024.0f,
+                             ((float32_t)(data->acc_z)) / 1024.0f};
+  float32_t amplitude =
+      sqrtf(scaled_imu[0] * scaled_imu[0] + scaled_imu[1] * scaled_imu[1] + scaled_imu[2] * scaled_imu[2]);
+  scaled_imu[0] /= amplitude;
+  scaled_imu[1] /= amplitude;
+  scaled_imu[2] /= amplitude;
+  /* Compute Quaternion from Accelerometer Data */
+  float32_t quat_meas_acc[4] = {0, 0, 0, 0};
```

Comment:
at lines 210 & 213 you're initializing with {0}. That should also be enough here.